interop/fake_grpclb/fake_grpclb.go

```diff
@@ -0,0 +1,178 @@
+	var err error
+	if lbReq, err = stream.Recv(); err != nil {
+		grpclog.Errorf("Error receiving LoadBalanceRequest: %v", err)
+		return err
+	}
+	grpclog.Info("LoadBalancerRequest received.")
+	initialReq := lbReq.GetInitialRequest()
+	if initialReq == nil {
+		grpclog.Info("Expected first request to be an InitialRequest. Got: %v", lbReq)
+		return status.Error(codes.Code(2), "First request not an InitialRequest")
+	}
+	// gRPC clients targetting foo.bar.com:443 can sometimes include the ":443" suffix in
+	// their requested names; handle this case. TODO: make 443 configurable?
+	cleanedName := initialReq.Name
+	if strings.HasSuffix(cleanedName, ":443") {
+		cleanedName = cleanedName[0 : len(cleanedName)-len(":443")]
+	}
+	if cleanedName != *serviceName {
+		grpclog.Info("Expected requested service name: %v. Got: %v", *serviceName, initialReq.Name)
+		return status.Error(codes.Code(2), "Bad requested service name")
+	}
+	if err := stream.Send(&lbpb.LoadBalanceResponse{
+		LoadBalanceResponseType: &lbpb.LoadBalanceResponse_InitialResponse{
+			InitialResponse: &lbpb.InitialLoadBalanceResponse{
+				ClientStatsReportInterval: &durationpb.Duration{
+					Seconds: 1,
+					Nanos:   0,
+				},
+			},
+		},
+	}); err != nil {
+		grpclog.Errorf("Error sending initial LB response: %v", err)
+		return status.Error(codes.Code(2), "Error sending initial response")
+	}
+	serverList := make([]*lbpb.Server, len(l.BackendAddrs))
+	for i := range l.BackendAddrs {
+		serverList[i] = &lbpb.Server{
+			IpAddress: l.BackendAddrs[i].Ip,
+			Port:      l.BackendAddrs[i].Port,
+		}
+	}
+	grpclog.Infof("Length of server list response is %d", len(serverList))
+	res := &lbpb.LoadBalanceResponse{
+		LoadBalanceResponseType: &lbpb.LoadBalanceResponse_ServerList{
+			ServerList: &lbpb.ServerList{
+				Servers: serverList,
+			},
+		},
+	}
+	grpclog.Info("Send LoadBalanceResponse: %v", res)
+	if err := stream.Send(res); err != nil {
+		grpclog.Errorf("Error sending LB response: %v", err)
+		return status.Error(codes.Code(2), "Error sending response")
+	}
+	if *shortStream {
+		return nil
+	}
+	for {
+		grpclog.Info("Send LoadBalanceResponse: %v", res)
+		if err := stream.Send(res); err != nil {
+			grpclog.Errorf("Error sending LB response: %v", err)
+			return status.Error(codes.Code(2), "Error sending response")
+		}
+		time.Sleep(10 * time.Second)
+	}
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	var opts []grpc.ServerOption
+	if *useTLS {
+		certFile := testdata.Path("server1.pem")
+		keyFile := testdata.Path("server1.key")
+		creds, err := credentials.NewServerTLSFromFile(certFile, keyFile)
+		if err != nil {
+			grpclog.Fatalf("Failed to generate credentials %v", err)
+		}
+		opts = append(opts, grpc.Creds(creds))
+	} else if *useALTS {
+		altsOpts := alts.DefaultServerOptions()
+		altsTC := alts.NewServerCreds(altsOpts)
+		opts = append(opts, grpc.Creds(altsTC))
+	}
+	rawBackendAddrs := strings.Split(*backendIps, ",")
+	if rawBackendAddrs[0] == "" {
```

Comment:
What does this mean? Is this a special case or just an error?

If this is an error, it's not necessary because the following parsing would report this as well.

Reply:
this was a messy way to handle the case when `--backend_addrs` is an empty string (which is actually valid and used for some test scenarios). Revised to clean this up